### PR TITLE
Add DeletePayment Integration Tests

### PR DESCRIPTION
**Pull request by enamourous**

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_deletePaymentsinglePerson_success() throws Exception {
```
Comment (sj-c) [inline src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java]: Great end-to-end coverage, both single and multi-person cases are well handled.

File: src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+/**
+ * Integration tests for {@code DeletePaymentCommand} interacting with the Model.
+ */
+public class DeletePaymentCommandIntegrationTest {
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
```
Comment (sj-c) [inline src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java]: Suggest adding an edge-case test later for deleting from an empty model — would ensure DeletePaymentCommand gracefully handles cases where no persons exist.

File: src/test/java/seedu/address/logic/parser/DeletePaymentCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public void parse_validArgs_returnsDeletePaymentCommand() throws Exception {
+        // Example: "1 p/1" → delete payment #1 from person #1
+        DeletePaymentCommand command = parser.parse("1 p/1");
+        assertEquals(new DeletePaymentCommand(List.of(Index.fromOneBased(1)), Index.fromOneBased(1)), command);
+    }
+
+    @Test
+    public void parse_multiplePersons_returnsDeletePaymentCommand() throws Exception {
+        // Example: "1,2 p/1" → delete payment #1 from persons #1 and #2
+        DeletePaymentCommand command = parser.parse("1,2 p/1");
```
Comment (dnt0n) [inline src/test/java/seedu/address/logic/parser/DeletePaymentCommandParserTest.java]: Would it be good to also add some test cases for unexpected formatting? For example, if the user enters leading/trailing spaces by accident, the program should be able to handle it. eg. deletepayment 1  ,2     ,3 p/    1

File: src/main/java/seedu/address/logic/parser/DeletePaymentCommandParser.java [hunk 4]
```diff
@@ -15,18 +15,34 @@ public class DeletePaymentCommandParser implements Parser<DeletePaymentCommand>
...
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_PAYMENT_INDEX);
 
-            List<Index> personIndexes = ParserUtil.parseIndexes(argMultimap.getPreamble());
-            Index paymentIndex = ParserUtil.parseIndex(argMultimap.getValue(CliSyntax.PREFIX_PAYMENT_INDEX)
-                    .orElseThrow(() -> new ParseException("Missing payment index.")));
+        List<Index> personIndexes;
+        Index paymentIndex;
 
-            return new DeletePaymentCommand(personIndexes, paymentIndex);
+        try {
```
Comment (jianronggu) [inline src/main/java/seedu/address/logic/parser/DeletePaymentCommandParser.java]: Excellent work at error handling here! Informative with the comments added :)

File: src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        // Prepare expected model
+        Person updatedPerson = personWithPayment.withRemovedPayment(payment);
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.setPerson(personWithPayment, updatedPerson);
+
+        String expectedMessage = String.format(DeletePaymentCommand.MESSAGE_SUCCESS, 1, updatedPerson.getName());
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+    }
+
+    @Test
```
Comment (jianronggu) [inline src/test/java/seedu/address/logic/commands/DeletePaymentCommandIntegrationTest.java]: Very clear testing with the guiding comments!
